src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.service.WriterService;
+import core.basesyntax.strategy.TransactionStrategy;
+
+public class ReportServiceImpl implements ReportService {
+    private final TransactionStrategy transactionStrategy;
+    private final WriterService writerService;
+
+    public ReportServiceImpl(TransactionStrategy transactionStrategy, WriterService writerService) {
+        this.transactionStrategy = transactionStrategy;
+        this.writerService = writerService;
+    }
+
+    @Override
+    public void countAmountOfFruits() {
+        Storage.fruitTransactions
+                .forEach(transaction -> transactionStrategy
+                        .get(transaction.getOperation())
+                        .doTransaction(transaction.getFruit(), transaction.getQuantity()));
+        StringBuilder stringBuilder = new StringBuilder("fruit,quantity" + System.lineSeparator());
```

Comment:
For the future, do not use such naming(stringBuilder, t, s) , how to guess what it stores?